### PR TITLE
feat: implement coverage report JSON mapper

### PR DESCRIPTION
**Pull request by zrock-dev**
## Related Tickets & US
- [CS-43-07 Coverage JSON deserealization](https://app.clickup.com/t/868b2dhv1)

## Description

Implementation of coverage report JSON mapper to Report object.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Implemented unit tests to verify feature behavior.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

- - -
This change is [Reviewable](https://reviewable.io/reviews/SoftDevSix/ArgosAPI/17)

### CODE REVIEW

File: src/test/java/com/softdevsix/api/domain/report/ReportTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    void verifyInvalidJSON(){
+        IReportReader reportReader = new JsonReportReader();
+        try {
+            reportReader.read("%n");
+            fail();
+        }catch (ReportReaderException e){
+            assertEquals("Could not read coverage file", e.getMessage());
+        }
+    }
+}
```
Comment (teran-joseluis) [inline src/test/java/com/softdevsix/api/domain/report/ReportTest.java]: Don't forget to add the extra line.

File: src/test/java/com/softdevsix/api/domain/report/ReportTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+class ReportTest {
+    public static String readFile(String path) {
```
Comment (teran-joseluis) [inline src/test/java/com/softdevsix/api/domain/report/ReportTest.java:17]: Don't you think this method should go into `JsonReportReader` class?

**Review by teran-joseluis — DISMISSED** [review]

**Review by salet-gutierrez-jalafund — APPROVED** [review]

File: src/test/java/com/softdevsix/api/domain/report/ReportTest.java (hunk 1 above)
Comment (zrock-dev) [reply src/test/java/com/softdevsix/api/domain/report/ReportTest.java]: This method is used only for testing purposes. The Filemanager HTTP client is fetching the JSON file as string.

File: src/test/java/com/softdevsix/api/domain/report/ReportTest.java (hunk 2 above)
Comment (zrock-dev) [reply src/test/java/com/softdevsix/api/domain/report/ReportTest.java:17]: Done

**Review by alex-paca-jalafund — APPROVED** [review]
Good work.

**Comment by zrock-dev** [comment]
We need this PR ASAP and I've required a review from a new reviewer. I'll proceed with the merge

**Pull request merged**